pkg/config/config.go

```diff
@@ -61,22 +60,23 @@ type IngressConfig struct {
 type MicroshiftConfig struct {
 	LogVLevel int `json:"logVLevel"`
 
-	SubjectAltNames []string `json:"subjectAltNames"`
-	NodeName        string   `json:"nodeName"`
-	NodeIP          string   `json:"nodeIP"`
-
-	Cluster ClusterConfig `json:"cluster"`
+	SubjectAltNames []string      `json:"subjectAltNames"`
+	NodeName        string        `json:"nodeName"`
+	NodeIP          string        `json:"nodeIP"`
+	BaseDomain      string        `json:"baseDomain"`
+	Cluster         ClusterConfig `json:"cluster"`
 
 	Ingress IngressConfig `json:"-"`
 }
 
-// Top level config
+// Top level config file
 type Config struct {
 	NodeName        string    `json:"nodeName"`
 	NodeIP          string    `json:"nodeIP"`
 	URL             string    `json:"url"`
-	ClusterDomain   string    `json:"clusterDomain"`
 	Network         Network   `json:"network"`
+	ClusterName     string    `json:"clusterName"`
```

Comment:
I don't see `ClusterName` being used in this PR. We had a discussion on slack about what the default for the cluster name should be, how to report it in the cluster, how to keep it from being changed, etc. so I think adding it will be a big enough change to deserve its own PR.

Reply:
Just as Doug pointed out, I do not see the usage of this `ClusterName` field. Perhaps, this is unintentional. I recommend removing it given it is not relevant to the changes in this PR.

Reply:
it seems to be needed in https://github.com/openshift/microshift/pull/1176/commits/5147e3fac8b87e1096d3fc0c51bc9788c003f87c#diff-a53aedd081647bc5a1bc5c1a969ad049f4ff08c9b4c5ba9408ea647488849234R281, but it is not used yet. I am also leaning towards adding that in a separate effort.

Reply:
We're not just replacing a name here, though. What used to be "ClusterDomain" will become "ClusterName.BaseDomain".

Considering this, is your suggestion then that BaseDomain should change its semantic between this PR and the "separate effort"? If not, the minimum we have to dl is to at least update all locations where this PR replaces ClusterDomain with BaseDomain to add a hardcoded ClusterName.

Reply:
Sorry it's unintentional, it was intended to be introduced on a PR after this one , which we needed to discuss.

Reply:
I am adding "microshift" for now as the clustername part, 
on a 2nd PR we can make that configurable.